tests/filecheck/backend/csl/print_csl.mlir

```diff
@@ -461,95 +451,67 @@ csl.func @builtins() {
 // CHECK-NEXT: const thing : imported_module = @import_module("<thing>");
 // CHECK-NEXT:
 // CHECK-NEXT: fn initialize() void {
-// CHECK-NEXT:   const lb : i16 = 0;
-// CHECK-NEXT:   const ub : i16 = 24;
-// CHECK-NEXT:   const step : i16 = 1;
-// CHECK-NEXT:   thing.some_func(lb, ub);
-// CHECK-NEXT:   const res : i32 = thing.some_func(lb, ub);
+// CHECK-NEXT:   thing.some_func(0, 24);
+// CHECK-NEXT:   const res : i32 = thing.some_func(0, 24);
 // CHECK-NEXT:   const v1 : comptime_struct = thing.some_field;
 // CHECK-NEXT:   const v2 : f32 = 3.14;
 // CHECK-NEXT:   const v0 : f16 = 2.718;
 // CHECK-NEXT:   const u32cst : u32 = 44;
 // CHECK-NEXT:
-// CHECK-NEXT:   for(@range(i16, lb, ub, step)) |idx| {
-// CHECK-NEXT:     const idx_f32 : f32 = @as(f32, idx);
-// CHECK-NEXT:     const idx_index : i32 = @as(i32, idx);
-// CHECK-NEXT:     A[idx_index] = idx_f32;
+// CHECK-NEXT:   for(@range(i16, 0, 24, 1)) |idx| {
+// CHECK-NEXT:     A[@as(i32, idx)] = @as(f32, idx);
 // CHECK-NEXT:   }
-// CHECK-NEXT:   const ub3 : i16 = 6;
 // CHECK-NEXT:
-// CHECK-NEXT:   for(@range(i16, lb, ub3, step)) |j| {
-// CHECK-NEXT:     const val : f32 = 1.0;
-// CHECK-NEXT:     const j_idx : i32 = @as(i32, j);
-// CHECK-NEXT:     x[j_idx] = val;
+// CHECK-NEXT:   for(@range(i16, 0, 6, 1)) |j| {
+// CHECK-NEXT:     x[@as(i32, j)] = 1.0;
 // CHECK-NEXT:   }
-// CHECK-NEXT:   const ub4 : i16 = 6;
 // CHECK-NEXT:
-// CHECK-NEXT:   for(@range(i16, lb, ub4, step)) |i| {
-// CHECK-NEXT:     const c2 : f32 = 2.0;
-// CHECK-NEXT:     const c0 : f32 = 0.0;
-// CHECK-NEXT:     const i_idx : i32 = @as(i32, i);
-// CHECK-NEXT:     b[i_idx] = c2;
-// CHECK-NEXT:     y[i_idx] = c0;
+// CHECK-NEXT:   for(@range(i16, 0, 6, 1)) |i| {
+// CHECK-NEXT:     b[@as(i32, i)] = 2.0;
+// CHECK-NEXT:     y[@as(i32, i)] = 0.0;
 // CHECK-NEXT:   }
 // CHECK-NEXT:   return;
 // CHECK-NEXT: }
 // CHECK-NEXT:
 // CHECK-NEXT: fn gemv() void {
-// CHECK-NEXT:   const lb : i32 = 0;
-// CHECK-NEXT:   const step : i32 = 1;
-// CHECK-NEXT:   const ub : i32 = 6;
-// CHECK-NEXT:   const ub1 : i32 = 4;
 // CHECK-NEXT:
-// CHECK-NEXT:   for(@range(i32, lb, ub1, step)) |i| {
-// CHECK-NEXT:     const tmp : f32 = 0.0;
-// CHECK-NEXT:     var tmp2 : f32 = tmp;
+// CHECK-NEXT:   for(@range(i32, 0, 4, 1)) |i| {
+// CHECK-NEXT:     var tmp : f32 = 0.0;
 // CHECK-NEXT:
-// CHECK-NEXT:     for(@range(i32, lb, ub, step)) |j| {
-// CHECK-NEXT:       const ix6 : i32 = i * ub;
-// CHECK-NEXT:       const ix6pj : i32 = ix6 +  j;
-// CHECK-NEXT:       const Axx : f32 = (A[ix6pj]) * (x[j]);
-// CHECK-NEXT:       tmp2 = tmp2 + Axx;
+// CHECK-NEXT:     for(@range(i32, 0, 6, 1)) |j| {
 // CHECK-NEXT:     }
-// CHECK-NEXT:     const tmp_plus_bi : f32 =  tmp2 + (b[i]);
-// CHECK-NEXT:     y[i] = tmp_plus_bi;
+// CHECK-NEXT:     y[i] = (tmp + (b[i]));
```

Comment:
Something is wrong here :thinking:

Reply:
Figured it out! It was the `scf.for` loop-carried variable trick from @dk949, should work now!